R/plotCCA.R

```diff
@@ -0,0 +1,540 @@
+      assays = matrix(ncol = nrow(object), dimnames = list(NULL, rownames(object))),
+      colData = DataFrame(row.names = rownames(object)),
+      reducedDims = list(RDA = object)
+    )
+    return(object)
+}
+
+# Get data for plotting
+#' @importFrom scater plotReducedDim retrieveCellInfo
+#' @importFrom SingleCellExperiment reducedDim reducedDimNames
+.incorporate_rda_vis <- function(
+        tse, dimred, ncomponents = 2, colour_by = color_by, color_by = NULL,
+        shape_by = NULL, size_by = NULL, order_by = NULL, text_by = NULL,
+        other_fields = list(), swap_rownames = NULL, point.padding = NA,
+        add_ellipse = TRUE, add_vectors = TRUE, add_significance = TRUE,
+        add_expl_var = TRUE, vec_lab = NULL, bins = NULL, ...){
+
+    # Check dimred
+    if( !dimred %in% reducedDimNames(tse) ){
+        stop("'dimred' must specify reducedDim.", call. = FALSE)
+    }
+    # Get reducedDim
+    reduced_dim <- reducedDim(tse, dimred)
+    # Check that there are at least 2 coordinates.
+    if( ncol(reduced_dim) < 2 ){
+        stop("reducedDim specified by 'dimred' must have at least 2 columns.", call. = FALSE)
+    }
+    # Only 2 dimensions are supported currently
+    ncomponents <- 2
+    # Make list of arguments for plotReducedDim
+    plotReducedDim_args <- list(
+      object = tse, dimred = dimred, ncomponents = ncomponents, colour_by = colour_by,
+      shape_by = shape_by, size_by = size_by, order_by = order_by, text_by = text_by,
+      text_size = 5, text_colour = "black", text_color = "black",
+      label_format = c("%s %i", " (%i%%)"), other_fields = other_fields,
+      swap_rownames = swap_rownames, point.padding = point.padding, force = 1,
+      rasterise = FALSE, scattermore = FALSE, summary_fun = "sum", hex = FALSE
+    )
+    # Get scatter plot with plotReducedDim --> keep theme similar between ordination methods
+    plot <- do.call("plotReducedDim", plotReducedDim_args)
+    
+    # Get data for ellipse
+    ellipse_data <- NULL
+    if( add_ellipse && !is.null(colour_by) ){
+        ellipse_data <- reduced_dim
+        ellipse_data <- as.data.frame(ellipse_data)
+        ellipse_data[[colour_by]] <- retrieveCellInfo(tse, colour_by)[["value"]]
+        attributes(ellipse_data)$colour_by <- colour_by
+    }
+    
+    # Get data for vectors
+    vector_data <- NULL
+    if( add_vectors ){
+        # Check if data is available
+        ind <- names(attributes(reduced_dim)) %in% c("rda", "cca")
+        # If it can be found
+        if( any(ind) ){
+            # Get biplot from cca object
+            rda <- attributes(reduced_dim)[ind][[1]]
+            vector_data <- rda$CCA$biplot
+            vector_data <- as.data.frame(vector_data)
+            vector_data[["group"]] <- rownames(vector_data)
+        } else{
+            # If it cannot be found, give warning
+            warning(
+                "CCA object was not found. Please calculate CCA by using runCCA.",
```

Comment:
The function also supports RDA and calculateRDA. Maybe "CCA/RDA object was not found. Please calculate it by using run*/calculate* function."